components/brave_wallet/browser/blockchain_registry_unittest.cc

```diff
@@ -557,46 +772,71 @@ TEST(BlockchainRegistryUnitTest, GetProvidersBuyTokens) {
   base::test::TaskEnvironment task_environment;
   auto* registry = BlockchainRegistry::GetInstance();
 
-  std::vector<mojom::BlockchainToken> buy_tokens;
-  for (const auto& v :
-       {GetRampBuyTokens(), GetSardineBuyTokens(), GetTransakBuyTokens()}) {
-    buy_tokens.insert(buy_tokens.end(), v.begin(), v.end());
-  }
+  absl::optional<OnRampTokensListMap> on_ramp_token_lists =
+      ParseOnRampTokensListMap(on_ramp_token_lists_json);
+  ASSERT_TRUE(on_ramp_token_lists);
+  registry->UpdateOnRampTokenLists(std::move(*on_ramp_token_lists));
 
-  const char* chains[] = {
-      mojom::kMainnetChainId,          mojom::kPolygonMainnetChainId,
-      mojom::kAvalancheMainnetChainId, mojom::kBinanceSmartChainMainnetChainId,
-      mojom::kSolanaMainnet,           mojom::kCeloMainnetChainId,
-      mojom::kArbitrumMainnetChainId};
-
-  for (auto* chain : chains) {
-    std::vector<mojom::BlockchainTokenPtr> expected_tokens;
-    for (const auto& token : buy_tokens) {
-      if (token.chain_id == chain) {
-        expected_tokens.push_back(mojom::BlockchainToken::New(token));
-      }
-    }
+  // Ethereum mainnet tokens are present
+  base::RunLoop run_loop;
+  registry->GetProvidersBuyTokens(
+      {mojom::OnRampProvider::kRamp, mojom::OnRampProvider::kSardine,
+       mojom::OnRampProvider::kTransak,
+       mojom::OnRampProvider::kTransak /* test duplicate provider */},
+      mojom::kMainnetChainId,
+      base::BindLambdaForTesting(
+          [&](std::vector<mojom::BlockchainTokenPtr> token_list) {
+            EXPECT_EQ(token_list.size(), 4UL);
+            run_loop.Quit();
+          }));
+  run_loop.Run();
 
-    base::RunLoop run_loop;
-    registry->GetProvidersBuyTokens(
-        {mojom::OnRampProvider::kRamp, mojom::OnRampProvider::kSardine,
-         mojom::OnRampProvider::kTransak,
-         mojom::OnRampProvider::kTransak /* test duplicate provider */},
-        chain,
-        base::BindLambdaForTesting(
-            [&](std::vector<mojom::BlockchainTokenPtr> token_list) {
-              EXPECT_EQ(expected_tokens, token_list) << chain;
+  // Polygon mainnet token is present
+  base::RunLoop run_loop2;
```

Comment:
please use `std::unique_ptr<base::RunLoop>` to prevent this kind of run loop naming. 
See `EthTxManagerUnitTest.SetGasPriceAndLimitForUnapprovedTransaction`